### PR TITLE
Add tests for the hub-and-spoke-vpns infra example

### PR DESCRIPTION
**Pull request by averbuks**
Hi @ludoo, it would be great to have your opinion, I'm pretty sure it can be cleaned up a little bit, also I'm not sure that we need to add some tests for `dns` related things

### CODE REVIEW

**Comment by averbuks** [comment]
Some not related tests are failing (most probably project services)

**Comment by ludoo** [comment]
> Some not related tests are failing (most probably project services)

Hmmm strange, I was pretty sure I fixed them all once we switched project version. I'll have a look soon.

File: tests/infrastructure/hub-and-spoke-vpns/test_cloud_routers.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# limitations under the License.
+
+"Test cloud routers resources creation in root module."
+
+
+def test_hub_custom_routers(plan):
+  "Custom routers should match input variables."
+  hub_custom_router_tpl = ('google_compute_router.hub-to-spoke-%s-custom[0]')
+  count = range(1, 3)
+  for i in count:
```
Comment (ludoo) [inline tests/infrastructure/hub-and-spoke-vpns/test_cloud_routers.py]: nit: can you skip the `count` variable?

File: tests/infrastructure/hub-and-spoke-vpns/test_cloud_routers.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    name = hub_custom_router_tpl % i
+    custom_router = plan.resource_changes[name]
+    assert custom_router['change']['after']['bgp'][0]['advertise_mode'] == 'CUSTOM'
+    assert custom_router['change']['after']['bgp'][0]['advertised_groups'] == ['ALL_SUBNETS']
+    assert custom_router['change']['after']['bgp'][0]['asn'] == plan.variables['hub_bgp_asn']
+    assert [range['range'] for range in custom_router['change']['after']['bgp'][0]['advertised_ip_ranges']] == [subnet['subnet_ip'] for subnet in plan.variables['spoke_%s_subnets' % (3 - i)]]
```
Comment (ludoo) [inline tests/infrastructure/hub-and-spoke-vpns/test_cloud_routers.py]: line is really too long: can you use a variable for `custom_router['change']['after']['bgp']`, break on parenthesis (use style guide or format using `autopep8` module in vscode), or break the line into a `for` loop?

File: tests/infrastructure/hub-and-spoke-vpns/test_cloud_routers.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+def test_spoke_routers(plan):
+  "Spoke routers should match input variables."
+  spoke_router_tpl = ('google_compute_router.spoke-%s')
+  spoke_bgp_asn_tpl = ('spoke_%s_bgp_asn')
+  count = range(1, 3)
+  for i in count:
```
Comment (ludoo) [inline tests/infrastructure/hub-and-spoke-vpns/test_cloud_routers.py]: nit: `count` variable

File: tests/infrastructure/hub-and-spoke-vpns/test_firewall.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+# limitations under the License.
+
+"Test firewall resources creation in root module."
+
+
+import pytest
+
+@pytest.fixture(scope='module')
+def firewall_modules(plan):
+  names = ['module.firewall-%s' %
```
Comment (ludoo) [inline tests/infrastructure/hub-and-spoke-vpns/test_firewall.py]: nit: I would filter for modules with `name.startswith('module.firewall-')` so if we change names the test won't break

**Comment by averbuks** [comment]
@ludoo not sure why the tests are failing, having `__init__.py` should prevent it, but it does not. 
also having `test_outputs.py` in foundations examples does not cause such a problem.

**Pull request merged**